nemo/collections/nlp/parts/nlp_overrides.py

```diff
@@ -391,7 +391,14 @@ def save_checkpoint(
             sharded_optim_state = self.optimizer_sharded_state_dict(
                 unsharded_optim_state=checkpoint['optimizer_states'][0]
             )
-            checkpoint['optimizer_states'] = [sharded_optim_state]
+
+            # Check whether to save optim states
```

Comment:
since we are not modifying storage_options maybe let's not turn it into a dict (just check if it's not None line below)